src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -34,219 +34,173 @@ public KestrelHttpParser(IKestrelTrace log)
             consumed = buffer.Start;
             examined = buffer.End;
 
-            ReadCursor end;
-            Span<byte> span;
-
-            // If the buffer is a single span then use it to find the LF
-            if (buffer.IsSingleSpan)
+            Span<byte> span = buffer.First.Span;
+            var lineIndex = span.IndexOfVectorized(ByteLF);
+            if (lineIndex >= 0)
             {
-                var startLineSpan = buffer.First.Span;
-                var lineIndex = startLineSpan.IndexOfVectorized(ByteLF);
-
-                if (lineIndex == -1)
-                {
-                    return false;
-                }
-
-                end = buffer.Move(consumed, lineIndex + 1);
-                span = startLineSpan.Slice(0, lineIndex + 1);
+                consumed = buffer.Move(consumed, lineIndex + 1);
+                span = span.Slice(0, lineIndex + 1);
             }
-            else
+            else if (buffer.IsSingleSpan || !GetMultiSpan(ref buffer, ref span, out consumed))
             {
-                var start = buffer.Start;
-                if (ReadCursorOperations.Seek(start, buffer.End, out end, ByteLF) == -1)
-                {
-                    return false;
-                }
-
-                // Move 1 byte past the \n
-                end = buffer.Move(end, 1);
-                var startLineBuffer = buffer.Slice(start, end);
-
-                span = startLineBuffer.ToSpan();
+                return false;
             }
 
             var pathStart = -1;
             var queryStart = -1;
             var queryEnd = -1;
             var pathEnd = -1;
-            var versionStart = -1;
 
             var httpVersion = HttpVersion.Unknown;
             HttpMethod method;
             Span<byte> customMethod;
             var i = 0;
             var length = span.Length;
-            var done = false;
 
             fixed (byte* data = &span.DangerousGetPinnableReference())
             {
-                switch (StartLineState.KnownMethod)
+                // Method
+                if (span.GetKnownMethod(out method, out var methodLength))
```

Comment:
Pointer version of version https://github.com/aspnet/KestrelHttpServer/pull/1463/commits/75a24af18aeeab45539cee85353c9f6ddf8ed789